cmd/fetchd/cmd/genesis-asi-upgrade.go

```diff
@@ -417,6 +434,54 @@ func ASIGenesisUpgradeWithdrawReconciliationBalances(jsonData map[string]interfa
 	return nil
 }
 
+func ASIGenesisUpgradeASISupply(jsonData map[string]interface{}, networkInfo NetworkConfig) {
+	denomInfo := networkInfo.DenomInfo
+	supplyInfo := networkInfo.SupplyInfo
+	additionalSupply, ok := sdk.NewIntFromString(supplyInfo.AdditionalSupplyValue)
+	if !ok {
+		panic("asi upgrade update supply: failed to convert new supply value to int")
+	}
+
+	if additionalSupply.LT(sdk.ZeroInt()) {
+		panic("asi upgrade update supply: additional supply value is negative")
+	}
+
+	bank := jsonData[banktypes.ModuleName].(map[string]interface{})
+	supply := bank["supply"].([]interface{})
+	balances := bank["balances"].([]interface{})
+	balancesMap := getGenesisBalancesMap(bank["balances"].([]interface{}))
+
+	var curSupply sdk.Int
+	var curSupplyIdx int
+	for idx, coin := range supply {
+		coinData := coin.(map[string]interface{})
+		if coinData["denom"] == denomInfo.NewDenom {
+			curSupplyIdx = idx
+			curSupply, ok = sdk.NewIntFromString(coinData["amount"].(string))
+			if !ok {
+				panic("asi upgrade update supply: failed to convert coin amount to int")
+			}
+			break
+		}
+	}
+
+	overflowAddressBalance := balances[(*balancesMap)[supplyInfo.UpdatedSupplyOverflowAddr]]
+	overflowAddressBalanceCoins := getCoinsFromInterfaceSlice(overflowAddressBalance)
+
+	additionalSupplyCoin := sdk.NewCoin(denomInfo.NewDenom, additionalSupply)
+	curSupplyCoin := sdk.NewCoin(denomInfo.NewDenom, curSupply)
+
+	// add new coins to the current supply
+	newSupplyCoins := additionalSupplyCoin.Add(curSupplyCoin)
```

Comment:
Just for consistency with add operation in line below, it would be better to swap operands in add operator:
 ```suggestion
	newSupplyCoins := curSupplyCoin.Add(additionalSupplyCoin)
```